MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -1285,7 +1285,7 @@ public boolean safeSpouse(Person p) {
                 !this.equals(p)
                 && !p.hasSpouse()
                 && p.oldEnoughToMarry()
-                && (!p.isPrisoner() || (p.isPrisoner() && isPrisoner()))
+                && (!p.isPrisoner() || isPrisoner())
```

Comment:
So this logic is we can marry the person if:
A. They are not a prisoner, or...
B.  we're a prisoner.

?

Reply:
That is correct. That enables people to go to the prisoner list and marry prisoners to personnel if they want to, but not the other way around.